lib/gui/modules/image-writer.js

```diff
@@ -124,7 +125,7 @@ imageWriter.service('ImageWriterService', function ($q, $rootScope) {
 
       if (error.code === 'EVALIDATION') {
         analytics.logEvent('Validation error', analyticsData)
-      } else if (error.code === 'EUNPLUGGED') {
+      } else if (BlockWriteStream.isTransientError(error)) {
```

Comment:
So, is a "transient error" always guaranteed to be a "drive unplugged" error?
Unplugging a drive doesn't sound very transient to me ;-)
I wonder if this `isTransientError` function should be moved outside of `BlockWriteStream`, given that a drive-unplugged event could also occur during validation, and not just during writing?

And apologies for the stupid question, but why do we have nearly identical error handling in both `image-writer.js` and `flash.js` ?

Reply:
Well, it's the same error codes you get if the device is unplugged, as they indicate a loss in connection (which is usually temporary, if the device hasn't been unplugged), hence transient.

Reply:
> And apologies for the stupid question, but why do we have nearly identical error handling in both image-writer.js and flash.js ?

I don't know; it's been a topic in meetings, that we need to consolidate our error handling, because there's quite a bit of duplication and it's scattered about. Not in the scope of this PR though.

Reply:
Ahhh, so I guess this is _effectively_ saying "if we had a transient error, and we still get the error even after 5 retries, then assume it means that the drive has been unplugged" ?